contracts/contracts/ocr2/SequencerUptimeFeed/library.cairo

```diff
@@ -0,0 +1,258 @@
+func RoundUpdated(status : felt, transmission_timestamp : felt):
+end
+
+@event
+func UpdateIgnored(
+    latest_status : felt, latest_timestamp : felt, incoming_status : felt, incoming_timestamp : felt
+):
+end
+
+@event
+func L1SenderTransferred(prev : felt, cur : felt):
+end
+
+@storage_var
+func s_l1_sender() -> (address : felt):
+end
+
+@storage_var
+func s_l2_cross_domain_messenger() -> (address : felt):
+end
+
+@storage_var
+func s_rounds(id : felt, field : felt) -> (res : felt):
+end
+
+@storage_var
+func s_latest_round_id() -> (res : felt):
+end
+
+func require_l1_sender{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    address : felt
+):
+    let (l1_sender) = s_l1_sender.read()
+    with_attr error_message("invalid sender"):
+        assert l1_sender = address
+    end
+
+    return ()
+end
+
+func require_valid_round_id{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    round_id : felt
+):
+    let (lateset_round_id) = s_latest_round_id.read()
+
+    with_attr error_message("invalid round_id"):
+        assert_not_zero(round_id)
+        # TODO: do we need to check if uint80 is overflown?
+        assert_le(round_id, lateset_round_id)
+    end
+
+    return ()
+end
+
+func require_access{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}():
+    let (address) = get_caller_address()
+    simple_read_access_controller.check_access(address)
+
+    return ()
+end
+
+# TODO: this methods are virtual in .sol: clarify
+@external
+func set_l1_sender{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    address : felt
+):
+    Ownable_only_owner()
+    _set_l1_sender(address)
+
+    return ()
+end
+
+@view
+func l1_sender{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}() -> (
+    address : felt
+):
+    let (address) = s_l1_sender.read()
+    return (address)
+end
+
+namespace sequencer_uptime_feed:
+    func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        initial_status : felt, owner_address : felt
+    ):
+        assert_boolean(initial_status)
+
+        simple_read_access_controller.constructor(owner_address)
+
+        let (timestamp) = get_block_timestamp()
+        _record_round(1, initial_status, timestamp)
+
+        return ()
+    end
+
+    func update_status{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        from_address : felt, status : felt, timestamp : felt
+    ):
+        alloc_locals
+        require_l1_sender(from_address)
+        assert_boolean(status)
+
+        let (latest_round_id) = s_latest_round_id.read()
+        let (latest_observation_timestamp) = s_rounds.read(
+            latest_round_id, Round.observation_timestamp
+        )
+        let (local latest_status) = s_rounds.read(latest_round_id, Round.answer)
+
+        let (lt) = is_le(timestamp, latest_observation_timestamp - 1)  # timestamp < latest_observation_timestamp
+        if lt == TRUE:
```

Comment:
can we just do `if lt` in Cairo instead of checking `== TRUE`?

Reply:
`lt` is also not used anywhere else so it might be cleaner to do

```
if is_le(timestamp, latest_observation_timestamp - 1):
```

instead of assigning `lt` to a variable.

Reply:
No :) even this isn't possible
`if is_le(timestamp, latest_observation_timestamp - 1) == TRUE`
cairo 💪